modules/core/perf/perf_allocation.cpp

```diff
@@ -0,0 +1,44 @@
+#include <array>
+#include "perf_precomp.hpp"
+
+using namespace perf;
+
+#define ALLOC_MAT_SIZES ::perf::szSmall24, ::perf::szSmall32, ::perf::szSmall64, \
+    ::perf::sz5MP, ::perf::sz2K, ::perf::szSmall128, ::perf::szODD, ::perf::szQVGA, \
+    ::perf::szVGA, ::perf::szSVGA, ::perf::sz720p, ::perf::sz1080p, ::perf::sz2160p, \
+    ::perf::sz4320p, ::perf::sz3MP, ::perf::szXGA, ::perf::szSXGA, ::perf::szWQHD, \
+    ::perf::sznHD, ::perf::szqHD
```

Comment:
BTW, It makes sense to make size as a test parameter and remove "types" (CV_8UC1, CV_16SC1, CV_8UC3, CV_8UC4).

Reply:
I don't think I can make this change.
I do want to test all sizes and all types. Unfortunately, the testing harness requires I pass one or more test parameters. And that parameter has a restricted list of possible c++ types. So I chose the CV_Types.

Reply:
Oooh. You want that static variable? That doesn't make sense. There is no logic to have it.

Reply:
And there is no "force" or negative from you, that's my poor choice of words. Better is, "if you require this as part of the PR, then I request...".

Here's my suggestion of the comment `// need for this static const is disputed; retaining as it doesn't cause any harm`